component/espejote-templates/rules-renderer-v1.libsonnet

```diff
@@ -0,0 +1,217 @@
+  for key in std.objectFields(o)
+  if std.isObject(o[key])
+} + {
+  [key]+: o[key]
+  for key in std.objectFields(o)
+  if std.isArray(o[key])
+} + {
+  [key]: o[key]
+  for key in std.objectFields(o)
+  if !std.isObject(o[key]) && !std.isArray(o[key])
+};
+
+local renderArray(arr) =
+  local realval(v) = std.lstripChars(v, '~');
+  local val_state = std.foldl(
+    function(a, it) a + it,
+    [
+      assert
+        std.isString(v) :
+        "renderArray() doesn't support arrays with non-string entries";
+      { [realval(v)]: !std.startsWith(v, '~') }
+      for v in arr
+    ],
+    {}
+  );
+  std.filter(
+    function(val) val_state[val],
+    std.objectFields(val_state)
+  );
+
+// Actual rendering
+
+// Apply this last to avoid problems with alert name matching
+local procPatchRules(group, config) =
+  group {
+    rules: [
+      if std.objectHas(rule, 'alert') then
+        rule {
+          // Change alert names so we don't get multiple alerts with the same name
+          // Ensure Watchdog is not renamed
+          alert: if super.alert == 'Watchdog' then 'Watchdog' else 'SYN_' + super.alert,
+          labels+: {
+            syn: 'true',
+            // mark alert as belonging to component instance in whose context the
+            // function is called.
+            syn_component: config.componentName,
+            // mark alert as belonging to the team in whose context the
+            // function is called.
+            [if config.teamLabel != null then 'syn_team']: config.teamLabel,
+          },
+        } + makeMergeable(std.get(config.patchRules, rule.alert, {}))
+      else
+        // If the rule doesn't have an alert field, return it unchanged (eg. recording rules)
+        rule
+      for rule in super.rules
+    ],
+  };
+
+local procPatchExprUserWorkload(group, config) =
+  local dropUserWorkload(alert) = std.member(config.ignoreUserWorkload, alert);
+  local patchThingy(field) = std.foldl(
+    function(a, b) b(a),
+    [
+      function(e)
+        std.strReplace(
+          e,
+          'namespace=~"openshift-monitoring|openshift-user-workload-monitoring"',
+          'namespace="openshift-monitoring"'
+        ),
+      function(e)
+        std.strReplace(
+          e,
+          'job=~"alertmanager-main|alertmanager-user-workload"',
+          'job="alertmanager-main"'
+        ),
+      function(e)
+        std.strReplace(
+          e,
+          'job=~"prometheus-k8s|prometheus-user-workload"',
+          'job="prometheus-k8s"'
+        ),
+    ],
+    field
+  );
+  group {
+    rules: [
+      if dropUserWorkload(std.get(rule, 'alert', '')) then
+        rule {
+          expr: patchThingy(super.expr),
+          // I have no clue what this part is originally for.
+          // I know what it does, just not why...
+          [if std.objectHas(std.get(rule, 'annotations', {}), 'description') then 'annotations']+: {
+            description: patchThingy(super.description),
+          },
```

Comment:
IIRC we wanted to update the description to not be misleading when we modify rules to not alert for the user workload stack. Essentially, we rewrite the description (if any) to not contain `openshift-user-workload-monitoring` here.